common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -84,7 +86,9 @@
                 public Object run() {
                     try {
                         final Field unsafeField = Unsafe.class.getDeclaredField("theUnsafe");
-                        Throwable cause = ReflectionUtil.trySetAccessible(unsafeField);
+                        // We always want to try using Unsafe as the access still works on java9 as well and
+                        // we need it for out native-transports and many optimizations.
```

Comment:
@normanmaurer have you looked into MethodHandles.byteBufferViewVarHandle [1] that creates a VarHandle that allows to do aligned/unaligned access on a ByteBuffer?

[1] http://download.java.net/java/jdk9/docs/api/java/lang/invoke/MethodHandles.html#byteBufferViewVarHandle-java.lang.Class-java.nio.ByteOrder-

http://mail.openjdk.java.net/pipermail/jigsaw-dev/2017-February/011477.html

Reply:
@johnou we are compiling against java8 so it will not help at all (and most people not use java9 yet).

Reply:
@normanmaurer perhaps for Netty 5 then given that Java 8 and 9 will be EOL this year unless you fork out cash http://www.oracle.com/technetwork/java/eol-135779.html

Reply:
@slandelle like this suggestion or is the public API not rich enough?